src/camp/generators/lib/campch.py

```diff
@@ -150,6 +151,18 @@ def as_text(ari:ari.ARI) -> str:
         enc.encode(ari, buf)
         return buf.getvalue()
 
+    def as_timeperiod(value:datetime) -> str:
+      seconds = value.total_seconds()
```

Comment:
I don't think that a `datetime` object has this method. A timedelta can be computed as a difference from the DTN time epoch (which is a constant in one of the ACE modules) and then the `total_seconds()` can be used. Does this filter function actually get exercised by jinja?

Reply:
Good point - `datetime` does not. I've updated the method and integrated `TP` into the unit testing.